source/collimation.f90

```diff
@@ -2180,94 +2122,132 @@ subroutine collimate_start
-            yj  = yj + half*c_length*ypj
-          end if
-        end if
+    gammax = (one + talphax(ie)**2)/tbetax(ie)
+    gammay = (one + talphay(ie)**2)/tbetay(ie)
 
-        gammax = (one + talphax(ie)**2)/tbetax(ie)
-        gammay = (one + talphay(ie)**2)/tbetay(ie)
+    do j=1,napx
+      xj  = (xv1(j)-torbx(ie))/c1e3
+      xpj = (yv1(j)-torbxp(ie))/c1e3
+      yj  = (xv2(j)-torby(ie))/c1e3
+      ypj = (yv2(j)-torbyp(ie))/c1e3
+      pj  = ejv(j)/c1e3
 
-        if(part_abs_pos(j).eq.0 .and. part_abs_turn(j).eq.0) then
-          nspx = sqrt(abs( gammax*(xj)**2 + two*talphax(ie)*xj*xpj +tbetax(ie)*xpj**2 )/myemitx0_collgap)
-          nspy = sqrt(abs( gammay*(yj)**2 + two*talphay(ie)*yj*ypj +tbetay(ie)*ypj**2 )/myemity0_collgap)
-          sum_ax(ie)   = sum_ax(ie) + nspx
-          sqsum_ax(ie) = sqsum_ax(ie) + nspx**2
-          sum_ay(ie)   = sum_ay(ie) + nspy
-          sqsum_ay(ie) = sqsum_ay(ie) + nspy**2
-          nampl(ie)    = nampl(ie) + 1
-        else
-          nspx = zero
-          nspy = zero
+      if(firstrun) then
+        if(iturn == 1 .and. j == 1) then
+          sum_ax(ie) = zero
+          sum_ay(ie) = zero
         end if
+      end if
 
-          sampl(ie)    = totals
-          ename(ie)    = bez(myix)(1:mNameLen)
-      end do !do j = 1, napx
-    end if !if(rselect.gt.0 .and. rselect.lt.65) then
-  end if !if( firstrun ) then
+      if(part_abs_pos(j) == 0 .and. part_abs_turn(j) == 0) then
+        nspx         = sqrt(abs(gammax*(xj)**2 + two*talphax(ie)*xj*xpj + tbetax(ie)*xpj**2)/myemitx0_collgap)
+        nspy         = sqrt(abs(gammay*(yj)**2 + two*talphay(ie)*yj*ypj + tbetay(ie)*ypj**2)/myemity0_collgap)
+        sum_ax(ie)   = sum_ax(ie) + nspx
+        sqsum_ax(ie) = sqsum_ax(ie) + nspx**2
+        sum_ay(ie)   = sum_ay(ie) + nspy
+        sqsum_ay(ie) = sqsum_ay(ie) + nspy**2
+        nampl(ie)    = nampl(ie) + 1
+      else
+        nspx = zero
+        nspy = zero
+      end if
+    end do
 
-!GRD HERE WE LOOK FOR ADEQUATE DATABASE INFORMATION
-  found = .false.
-  if(cdb_elemMap(myix) > 0) then
-    icoll = cdb_elemMap(myix)
-    found = .true.
   end if
 
-  if(.not. found .and. firstrun .and. iturn.eq.1) then
-    write(lout,"(a)") "COLL> WARNING Collimator not found in colldb: '"//trim(bez(myix))//"'"
+end subroutine collimate_trackThin
+
+! ================================================================================================ !
+!  Updated: 2019-09-12
+!  Called when we enter a collimator.
+!  The routine doesn't check that the collimator exists! This should be checked before calling it.
+! ================================================================================================ !
+subroutine collimate_start_collimator(stracki)
+
+  use coll_db
+  use mod_common
+  use mod_common_main
+  use mod_common_track
+
+  real(kind=fPrec), intent(in) :: stracki
+
+  integer j
+
+  icoll    = cdb_elemMap(myix)
+  nsig     = cdb_cNSig(icoll)
+  c_length = zero
```

Comment:
I think that the actual `collimate_start_collimator` subroutine should finish here.
The following part should be some kind of `collimate_stats`

Reply:
The very last step in the line of steps will be reorganising the main collimation module. I don't want to make any major changes to the order of things until then.